.github/workflows/ci.yml

```
@@ -264,6 +264,26 @@ jobs:
         rustup target add thumbv6m-none-eabi
         cargo check -p cordyceps --target thumbv6m-none-eabi
 
+  # run no-atomics target tests
+  maitake_sync_no_atomics_compat:
+    needs: changed_paths
+    if: needs.changed_paths.outputs.should_skip != 'true' || !fromJSON(needs.changed_paths.outputs.paths_result).maitake.should_skip
+    runs-on: ubuntu-latest
+    name: no-atomics compat (maitake-sync)
+    steps:
+    - name: install rust toolchain
+      run: rustup show
+    - uses: actions/checkout@v4
+    - name: run check
+      # we add the target HERE instead of in the 'install rust toolchain' step
+      # because we need the `rust-toolchain.toml` file to be checked out, otherwise
+      # the target is only added for `stable`. We could also add this to the
+      # `targets` item in the toolchain file, but that would be a waste for all
+      # other tests.
+      run: |
+        rustup target add thumbv6m-none-eabi
+        cargo check -p maitake-sync --target thumbv6m-none-eabi --no-default-features --features=critical-section
```

hmm, didn't we previously determine that [the `install rust toolchain` step is Just Wrong Everywhere](https://github.com/hawkw/mycelium/issues/528) since it runs before the toolchain file is checked out? we should fix that everywhere...but in the meantime, I think we should do the following here:

- change this so that there's a separate toolchain step that just installs `thumbv6m-none-eabi` after checkout
- edit the comment to just say "we don't install the toolchain in the toolchain file because we don't need it for every build"

```suggestion
    steps:
    - uses: actions/checkout@v4
    - name: install rust toolchain
      # We add the target here, rather than in the `targets` item in the 
      # toolchain file, because it would be a waste for all other CI jobs to
      # have to install it.
      run: rustup target add thumbv6m-none-eabi
    - name: run check
      run: cargo check -p maitake-sync --target thumbv6m-none-eabi --no-default-features --features=critical-section
```